extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/internal/job/DailyJobSun.java

```diff
@@ -7,59 +7,127 @@
  */
 package org.eclipse.smarthome.binding.astro.internal.job;
 
+import static com.google.common.base.Preconditions.checkNotNull;
 import static org.eclipse.smarthome.binding.astro.AstroBindingConstants.*;
+import static org.eclipse.smarthome.binding.astro.internal.job.Job.*;
+import static org.eclipse.smarthome.binding.astro.internal.model.SunPhaseName.*;
 
 import org.eclipse.smarthome.binding.astro.handler.AstroThingHandler;
 import org.eclipse.smarthome.binding.astro.internal.model.Planet;
 import org.eclipse.smarthome.binding.astro.internal.model.Sun;
 import org.eclipse.smarthome.binding.astro.internal.model.SunEclipse;
-import org.eclipse.smarthome.binding.astro.internal.model.SunPhaseName;
 
 /**
- * Schedules the events for the sun for the current day.
+ * Daily Scheduled Jobs For Sun Planet
  *
  * @author Gerhard Riegler - Initial contribution
+ * @author Amit Kumar Mondal - Implementation to be compliant with ESH Scheduler
  */
-public class DailyJobSun extends AbstractDailyJob {
+public final class DailyJobSun implements Job {
 
+    private final String thingUID;
+    private final AstroThingHandler handler;
+
+    /**
+     * Constructor
+     *
+     * @param thingUID
+     *            the Thing UID
+     * @param handler
+     *            the {@link AstroThingHandler} instance
+     * @throws NullPointerException
+     *             if {@code thingUID} or {@code handler} is {@code null}
+     */
+    public DailyJobSun(String thingUID, AstroThingHandler handler) {
+        checkNotNull(thingUID, "Thing UID cannot be null");
```

Comment:
Please change it to a normal `null` check and throw a `IllegalArgumentException`. This is more meaningful than throwing an NPE here.